internal/olfactor/olfactor_test.go

```diff
@@ -0,0 +1,55 @@
+package olfactor_test
+
+import (
+	"io"
+	"testing"
+
+	"github.com/buildkite/agent/v3/internal/olfactor"
+	"gotest.tools/v3/assert"
+)
+
+func TestOlfactor(t *testing.T) {
```

Comment:
Most of `olfactor`s string decoding should be tested in the `replacer` package, so these tests are fairly simple.